src/Function/Metric/Rational/CauchySequence.agda

```diff
@@ -0,0 +1,83 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Cauchy sequences on a metric over the rationals
+------------------------------------------------------------------------
+
+{-# OPTIONS --safe --cubical-compatible --guardedness #-}
+
+open import Function.Metric.Rational.Bundles
+
+module Function.Metric.Rational.CauchySequence {a ℓ} (M : Metric a ℓ) where
+
+open Metric M hiding (_≈_)
+
+open import Codata.Guarded.Stream
+open import Codata.Guarded.Stream.Properties
+open import Data.Nat.Base as ℕ using (ℕ; z≤n; s≤s)
+import Data.Nat.Properties as ℕ
+open import Data.Integer.Base using (+<+)
+open import Data.Product.Base
+open import Data.Rational.Base
+open import Data.Rational.Properties
+open import Function.Base
+open import Level
+open import Relation.Binary
+open import Relation.Binary.PropositionalEquality using (cong₂)
+
+record CauchySequence : Set a where
```

Comment:
So I think this and the equality definition needs to be defined over a `DistanceFunction` rather than a full `Metric` with laws. Otherwise we're forced to define completeness over the whole metric space bundle, and we try and avoid defining predicates over bundles like that in the library.

Reply:
The added advantage of this, is that we could definite the notion of Cauchy sequence generically, rather than just for `Rational`.

Reply:
This record would also benefit from a constructor I think? Which would allow us to avoid writing `sequence x` and `sequence y` everywhere and instead write `x` and `y`?

Reply:
Hmm. I want to make `isCauchy` irrelevant. Would both doing that and introducing a constructor run into Irrelevant Projections stuff?

Reply:
Well we use a record constructor with irrelevant fields in Data.Rational.Base and I'm not aware we've run into any problems there...

Reply:
As for irrelevant projections, the discussion on https://github.com/agda/agda/issues/7063 , upstream from #2199 (and others `label:irrelevance`/`label:irrelevant-projections`)  suggest... (probably) not?